src/Compilers/Shared/BuildServerConnection.cs

```diff
@@ -96,7 +96,7 @@ internal sealed class BuildServerConnection
                         try
                         {
                             var process = Process.GetProcessById(shutdownBuildResponse.ServerProcessId);
-#if NET50_OR_GREATER
+#if NET5_0_OR_GREATER
```

Comment:
@jaredpar how broken was our code because of this?

Reply:
@333fred The intent is that both `WaitForExit` and `WaitForExitAsync` are semantically equivalent (https://github.com/dotnet/runtime/issues/34294#issuecomment-606833721).

However, I found https://github.com/dotnet/runtime/issues/42556, which makes me less confident that we want to call the async version in .NET 5.0. It's a bug where they're not semantically equivalent and it was fixed for .NET 6, but not ported to .NET 5.

Reply:
In this particular case the distinction should not matter because we aren't actually reading output in the shutdown path. It's more just "wait for this op to complete".

Reply:
In terms of "how broken" the answer is "not really that broken". It's mostly about using `Async` when possible to avoid blocking threads. This is just on the "jaredpar is embarrased" level than bad scenario.